Shared/Supporting Files/Views/DownloadOfflineResourcesView.swift

```diff
@@ -0,0 +1,172 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+/// A view with controls for downloading the app's on-demand resources.
+struct DownloadOfflineResourcesView: View {
+    /// The action to dismiss the view.
+    @Environment(\.dismiss) private var dismiss
+    
+    /// The view models for the on-demand resource requests.
+    @State private var onDemandResources = SamplesApp.samples.compactMap { sample in
+        sample.hasDependencies ? OnDemandResource(sample: sample) : nil
+    }
+    
+    /// The current state of the "download all on-demand resources" request.
+    @State private var downloadAllRequestState: OnDemandResource.RequestState?
```

Comment:
By making the initializer for `OnDemandResource` asynchronous, the request state would no longer need to be optional.